src/mappings/token/utils.ts

```diff
@@ -306,3 +319,41 @@ export async function processTokenMetadata(
     }
   }
 }
+
+export async function getHolderAccountsForToken(
+  em: EntityManager,
+  tokenId: string
+): Promise<Account[]> {
+  const holders = await em.getRepository(TokenAccount).findBy({ tokenId })
+
+  const holdersMemberIds = holders
+    .filter((follower) => follower?.memberId)
+    .map((follower) => follower.memberId as string)
+
+  const limit = pLimit(10) // Limit to 10 concurrent promises
+  const holdersAccounts: (Account | null)[] = await Promise.all(
+    holdersMemberIds.map((membershipId) =>
+      limit(async () => await em.getRepository(Account).findOneBy({ membershipId }))
+    )
+  )
+
+  return holdersAccounts.filter((account) => account) as Account[]
+}
+
+export async function notifyTokenHolders(
+  em: EntityManager,
+  tokenId: string,
+  notificationType: NotificationType,
+  event?: Event
+) {
+  const holdersAccounts = await getHolderAccountsForToken(em, tokenId)
+  for (const holdersAccount of holdersAccounts) {
+    await addNotification(
+      em,
+      holdersAccount,
+      new MemberRecipient({ membership: holdersAccount.membershipId }),
+      notificationType,
+      event
+    )
+  }
+}
```

Comment:
Shouldn't this be done using `promise.all`, as it could take a considerable time if token has large number of holders. Or better use `pLimit`

Reply:
👍